### PR TITLE
feat: create new axios instance for each composio instance

### PR DESCRIPTION
**Pull request by utkarsh-dixit**
>[!IMPORTANT]

>Each `Composio` instance now creates its own `axios` instance, allowing multiple clients with different API keys, with tests verifying this behavior.

- **Behavior**:- Each `Composio` instance now creates its own `axios` instance in `index.ts`, allowing multiple clients with different API keys.
    - Tests in `index.spec.ts` ensure different `axios` instances for different clients and correct API key usage.
- **Models**:- Updated constructors in `Entity.ts`, `actions.ts`, `activeTriggers.ts`, `apps.ts`, `connectedAccounts.ts`, `integrations.ts`, and `triggers.ts` to accept `axios` instance.
    - `AxiosBackendClient` in `backendClient.ts` now uses `createClient` from [hey-api](https://github.com/hey-api)`/client-axios`.
- **Tests**:- Added test in `index.spec.ts` to verify multiple clients with different API keys.
    - Updated tests in `Entity.spec.ts`, `actions.spec.ts`, `activeTriggers.spec.ts`, `apps.spec.ts`, `connectedAccounts.spec.ts`, `integrations.spec.ts`, and `triggers.spec.ts` to accommodate new `axios` instance handling.

<sup>This description was created by </sup>

<[https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral%7C<https://img.shields.io/badge/Ellipsis-blue?color=175173%7CEllipsis>><sup](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&amp;utm_source=github&amp;utm_medium=referral%7C&lt;https://img.shields.io/badge/Ellipsis-blue?color=175173%7CEllipsis&gt;&gt;&lt;sup)> for 9b9a8b087ae58ecc8046fb4436e8ef43279b742b. It will automatically update as commits are pushed.</sup>

----

> [!IMPORTANT]
> Each `Composio` instance now creates its own `axios` instance, allowing multiple clients with different API keys, with tests verifying this behavior.
> 
>   - **Behavior**:
>     - Each `Composio` instance now creates its own `axios` instance in `index.ts`, allowing multiple clients with different API keys.
>     - Tests in `index.spec.ts` ensure different `axios` instances for different clients and correct API key usage.
>   - **Models**:
>     - Updated constructors in `Entity.ts`, `actions.ts`, `activeTriggers.ts`, `apps.ts`, `connectedAccounts.ts`, `integrations.ts`, and `triggers.ts` to accept `axios` instance.
>     - `AxiosBackendClient` in `backendClient.ts` now uses `createClient` from `@hey-api/client-axios`.
>   - **Tests**:
>     - Added test in `index.spec.ts` to verify multiple clients with different API keys.
>     - Updated tests in `Entity.spec.ts`, `actions.spec.ts`, `activeTriggers.spec.ts`, `apps.spec.ts`, `connectedAccounts.spec.ts`, `integrations.spec.ts`, and `triggers.spec.ts` to accommodate new `axios` instance handling.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for 75c77e4f68a2a0d78b52fdbab7dcd228ea9fb0a5. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by himanshu-dixit — APPROVED** [review]

File: js/src/sdk/models/backendClient.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@
    * The runtime environment where the client is being used.
    */
   public runtime: string;
-  public instance: AxiosInstance;
+  public instance: Client;
```
Comment (shreysingla11) [inline js/src/sdk/models/backendClient.ts:25]: Consider adding JSDoc for the instance property to improve code documentation:
```typescript
/**
 * The axios client instance specific to this backend client.
 * Each Composio instance gets its own axios instance to prevent
 * configuration conflicts between multiple SDK instances.
 */
public instance: Client;
```

File: js/src/sdk/models/backendClient.ts [hunk 2]
```diff
@@ -72,33 +78,16 @@
...
-
-    setAxiosClientConfig(axiosClient.instance);
-    this.instance = axiosClient.instance;
-  }
-
   getAxiosInstance() {
-    return axiosClient.instance;
+    return this.instance.instance;
   }
 }
```
Comment (shreysingla11) [inline js/src/sdk/models/backendClient.ts:91]: Consider adding a cleanup method to properly dispose axios instances when no longer needed:
```typescript
public dispose(): void {
  if (this.instance) {
    // Cancel any pending requests
    this.instance.cancelAllRequests();
    // Clear any internal state
    this.instance = null;
  }
}
```

**Comment by shreysingla11** [comment]
I've reviewed the changes and here's my assessment:

**Positive Aspects:**
1. Good separation of concerns by creating instance-specific axios clients
2. Improved type safety with better TypeScript usage
3. Added proper tests to verify multiple client instances work correctly
4. Better dependency injection pattern

**Suggestions for Improvement:**
1. Add error handling for client creation (see comment on backendClient.ts)
2. Add proper JSDoc documentation for the instance property
3. Add a cleanup method to properly dispose axios instances
4. Consider adding more edge case tests:
   - Test error scenarios
   - Test with invalid configurations
   - Test concurrent requests with multiple instances

**Overall Assessment:**
The changes represent a significant improvement in the SDK's architecture by removing global state dependencies and improving isolation between different SDK instances. The code is well-structured and follows good practices.

Rating: 8/10 - Good quality changes with room for minor improvements in error handling and documentation.
